Library/Homebrew/language/python.rb

```diff
@@ -96,5 +100,84 @@ def self.setup_install_args(prefix)
     def self.package_available?(python, module_name)
       quiet_system python, "-c", "import #{module_name}"
     end
+
+    def self.virtualenv(formula, venv_root, python="python")
+      venv = Virtualenv.new venv_root, python
+      # Do some cute metaprogramming so that when `system` is invoked by the
+      # Virtualenv instance, `Formula#system` is called, so that we get the
+      # output capture and logging behavior we expect from formulae.
```

Comment:
I think that's probably a good reason to just pull `Formula#system` into a utility function; there's been a few other places where we've wanted something similar.